misc-utils/lsfd-unkn.c

```diff
@@ -551,6 +552,165 @@ static const struct anon_ops anon_eventpoll_ops = {
 	.attach_xinfo = anon_eventpoll_attach_xinfo,
 };
 
+/*
+ * timerfd
+ */
+struct anon_timerfd_data {
+	int clockid;
+	struct itimerspec itimerspec;
+};
+
+static bool anon_timerfd_probe(const char *str)
+{
+	return strncmp(str, "[timerfd]", 9) == 0;
+}
+
+static void anon_timerfd_init(struct unkn *unkn)
+{
+	unkn->anon_data = xcalloc(1, sizeof(struct anon_timerfd_data));
+}
+
+static void anon_timerfd_free(struct unkn *unkn)
+{
+	struct anon_timerfd_data *data = unkn->anon_data;
+	free(data);
+}
+
+static int anon_timerfd_handle_fdinfo(struct unkn *unkn, const char *key, const char *value)
+{
+	struct anon_timerfd_data *data = (struct anon_timerfd_data *)unkn->anon_data;
+
+	if (strcmp(key, "clockid") == 0) {
+		unsigned long clockid;
+		char *end = NULL;
+
+		errno = 0;
+		clockid = strtoul(value, &end, 0);
+		if (errno != 0)
+			return 0; /* ignore -- parse failed */
+		if (*end != '\0')
+			return 0; /* ignore -- garbage remains. */
+
+		data->clockid = clockid;
+		return 1;
+	} else if (strcmp(key, "it_value") == 0 ||
+		   strcmp(key, "it_interval") == 0) {
+		uint64_t tv_sec;
+		uint64_t tv_nsec;
+
+		if (sscanf(value, "(%"SCNu64", %"SCNu64")",
+			   &tv_sec, &tv_nsec) == 2) {
+			struct timespec *t = (key[3] == 'v')
```

Comment:
This check is a bit underhanded. A full strcmp would be clearer.

Reply:
To reduce calling strcmp, I changed the control flow.
The underhanded code is deleted.